mongodbatlas/resource_mongodbatlas_federated_database_instance.go

```diff
@@ -661,6 +661,7 @@ func newDataFederationDataSource(collectionFromConf map[string]interface{}) []ad
 			Database:            stringPtr(dataSourceFromConfMap["database"].(string)),
 			Collection:          stringPtr(dataSourceFromConfMap["collection"].(string)),
 			CollectionRegex:     stringPtr(dataSourceFromConfMap["collection_regex"].(string)),
+			DatabaseRegex:       stringPtr(dataSourceFromConfMap["database_regex"].(string)),
```

Comment:
I suggest to create/modify a regression test that would fail before making this change.

Reply:
good point @lantoli  --- added test coverage and quickly verified with TDD.